### ISSUE TITLE
Dependency to deprecated / EOL code-coverage-api

### ISSUE THREAD

**Issue by offa** [jenkinsci/cobertura-plugin#219]
### Jenkins and plugins versions report

any

### What Operating System are you using (both controller, and any agents involved in the problem)?

any

### Reproduction steps

a. See Jenkins UI warning
b. See code-coverage-api project (eg. https://github.com/jenkinsci/code-coverage-api-plugin)

### Expected Results

Migration to [Coverage Plugin](https://github.com/jenkinsci/coverage-plugin) (as suggested in the docs)

### Actual Results

Deprecated / EOL dependency is used

### Anything else?

The new Coverage plugin seems not to be a drop in replacement and code changes are likely needed.

**Comment by jamiejackson** [comment]
Maybe you'd have better luck on the [Jira tracker](https://www.jenkins.io/participate/report-issue/redirect/#15500). I'm not sure which tracker gets more attention.

**Comment by jgournet** [comment]
Someone created a ticket: https://issues.jenkins.io/browse/JENKINS-73238 (thank you !).
Their answer was: functionality from cobertura plugin should be included in "coverage" ... so we can just remove it

**Comment by jamiejackson** [comment]
Thanks, @jgournet . Let us know what happens if and when you try it out.

**Comment by offa** [comment]
Indeed, Coverage is a better choice. Let's close here then.

**Comment by offa** [comment]
There's a new [Coverage](https://plugins.jenkins.io/coverage/) plugin, not to confuse with the _Code Coverage_ Plugin.

**Comment by captain-caveman2k** [comment]
> There's a new [Coverage](https://plugins.jenkins.io/coverage/) plugin, not to confuse with the _Code Coverage_ Plugin.

Yep, I understand that and I have that installed in my Jenkins installation:

![image](https://github.com/jenkinsci/cobertura-plugin/assets/1523871/026cd442-10e1-481d-9916-a6d3bc086e8d)

Unfortunately I cannot uninstall or disable Code Coverage v4.99 as Cobertura v1.17 has a dependency on it - unless there is another way of performing that. If so could you please enlighten me?

**Comment by offa** [comment]
With the new Coverage plugin you should be able to replace the Cobertura and (old) Coverage plugin.

**Comment by captain-caveman2k** [comment]
> With the new Coverage plugin you should be able to replace the Cobertura and (old) Coverage plugin.

Thanks - Like others I thought the new Coverage plug-in was a replacement for the old Code Coverage plug-in rather than a replacement for both.

I have now updated my Jenkinsfile to use `recordCoverage `and uninstalled both the Cobertura plug-in and the old Code Coverage plug-in.

**Comment by ddalcino** [comment]
> I have now updated my Jenkinsfile to use `recordCoverage` instead and uninstalled both the Cobertura plug-in and the old Code Coverage plug-in.

Thank you; I have done the same.

Just in case anyone is unclear on what changes are required to make `Coverage` work properly, I will share what I did here. I used the pipeline snippet generator for `recordCoverage` (go to 'Pipeline Syntax > Snippet Generator') to generate the necessary code. After modifying my Jenkinsfile accordingly, everything is working properly. Here's the diff for my Jenkinsfile:

```diff
- cobertura(coberturaReportFile: 'coverage/coverage.xml', onlyStable: false)
+ recordCoverage(tools: [[parser: 'COBERTURA', pattern: 'coverage/coverage.xml']])
```

The `onlyStable: false` k-v pair is unnecessary for Coverage; it is automatically recording coverage for both stable and unstable builds.
